channeld/channeld.c

```diff
@@ -2199,40 +2586,1304 @@ static void handle_unexpected_reestablish(struct peer *peer, const u8 *msg)
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Unable to calculate intiator fee.");
+
+	/* Calculate accepter fee contribution */
+	if (!amount_sat_sub(&accepter_fee, accepter_in, accepter_out))
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Unable to calculate accepter fee");
+	/* An extra check for cleaner log messages */
+	if (amount_sat_less(accepter_fee, peer->splice_accepter_funding)) {
+		msg = towire_channeld_splice_funding_error(NULL, accepter_fee,
+							   peer->splice_accepter_funding,
+							   false);
+		wire_sync_write(MASTER_FD, take(msg));
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Accepter funding is less than commited "
+				 "amount. Accepter adding %s and taking %s out "
+				 " and they committed to %s.",
+				 fmt_amount_sat(tmpctx, accepter_in),
+				 fmt_amount_sat(tmpctx, accepter_out),
+				 fmt_amount_sat(tmpctx, peer->splice_accepter_funding));
+	}
+	if (!amount_sat_sub(&accepter_fee, accepter_fee,
+			    peer->splice_accepter_funding))
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Unable to calculate accepter fee.");
+
+	/* Check intiator fee */
+	if (amount_sat_less(initiator_fee, min_initiator_fee)) {
+		msg = towire_channeld_splice_feerate_error(NULL, initiator_fee,
+							   false);
+		wire_sync_write(MASTER_FD, take(msg));
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "%s fee was too low", opener ? "Our" : "Your");
+	}
+	if (!peer->splice_force_feerate && opener
+		&& amount_sat_greater(initiator_fee, max_initiator_fee)) {
+		msg = towire_channeld_splice_feerate_error(NULL, initiator_fee,
+							   true);
+		wire_sync_write(MASTER_FD, take(msg));
+		/* DTODO: Swap `peer_failed_warn` out for `tx_abort` */
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Our own fee was too high");
+	}
+	/* Check accepter fee */
+	if (amount_sat_less(accepter_fee, min_accepter_fee)) {
+		msg = towire_channeld_splice_feerate_error(NULL, accepter_fee,
+							   false);
+		wire_sync_write(MASTER_FD, take(msg));
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "%s fee was too low", opener ? "Your" : "Our");
+	}
+	if (!peer->splice_force_feerate && !opener
+		&& amount_sat_greater(accepter_fee, max_accepter_fee)) {
+		msg = towire_channeld_splice_feerate_error(NULL, accepter_fee,
+							   true);
+		wire_sync_write(MASTER_FD, take(msg));
+		/* DTODO: Swap `peer_failed_warn` out for `tx_abort` */
+		peer_failed_warn(peer->pps, &peer->channel_id,
+				 "Our own fee was too high");
+	}
+
+	return funding_amount;
+}
+
+static void reset_splice(struct peer *peer)
+{
+	peer->splice_opener_funding = AMOUNT_SAT(0);
+	peer->splice_accepter_funding = AMOUNT_SAT(0);
+	peer->splice_feerate_per_kw = 0;
+	peer->splice_force_feerate = false;
+	peer->splice_force_sign_first = false;
+	peer->on_stfu_success = NULL;
+	peer->tx_add_input_count = 0;
+	peer->tx_add_output_count = 0;
+	peer->current_psbt = NULL;
+	peer->received_tx_complete = false;
```

Comment:
it might be nicer to bundle all of the state pieces for a splice into a single new struct off peer that you can auto-matically reset?

Reply:
Done and done! Feels much cleaner now.